go/mysql/auth_server_static.go

```diff
@@ -205,15 +206,15 @@ func (a *AuthServerStatic) Salt() ([]byte, error) {
 // ValidateHash is part of the AuthServer interface.
 func (a *AuthServerStatic) ValidateHash(salt []byte, user string, authResponse []byte, remoteAddr net.Addr) (Getter, error) {
 	// If enough time has elpased (default is 5 minutes), reload the auth file/string
-	if time.Now().Sub(a.LastAuthRotation) > *mysqlAuthRotationTime {
-		a.loadConfigFromParams(*mysqlAuthServerStaticFile, *mysqlAuthServerStaticString)
-		a.LastAuthRotation = time.Now()
+	lastAuth := a.LastAuthRotation
+	if (time.Now().Unix() - lastAuth) > *mysqlAuthRotationTime {
+		if swapped := atomic.CompareAndSwapInt64(&a.LastAuthRotation, lastAuth, time.Now().Unix()); swapped {
```

Comment:
It seems to me the whole thing would be simplified if we just spun up a `Timer` instance to re-read the config periodically rather than trying to load it on-demand.

That way the regular accessor threads don't have to worry about checking the update time.

It might require adding a simple Stop() interface to the AuthServer so we can stop the timer at shutdown time, but that seems pretty straightforward IMO and would be an improvement.

Reply:
I'm thinking we should do a uniform scheme that applies to all files. Currently, some are reloaded through a SIGHUP. So, I'd say let's do the SIGHUP handling for this definitely.

The next question: will a timer based reload be still be useful once we have SIGHUP? Not sure. What do you think?

Reply:
Oh that's a good point.

In fact looking at the code, the mysql auth server itself handles SIGHUP already, so maybe this whole effort should be rolled back?

Reply:
This makes sense, I've rolled back the auth rotation changes and changed the mutex back to a normal one.